docs/advanced/tls_porting.md

```diff
@@ -24,46 +24,56 @@ mbed TLS distinguishes between strong and weak entropy sources. Of the sources r
 
 The preferred way to provide a custom entropy source:
 
-1. Implement the `mbedtls_hardware_poll` function to let it access the device's entropy source.
-2. Set the appropriate macros for your target in `hal/targets.json`.
+1. Implement the `trng_api.h` to let mbed TLS access the device's entropy source.
+2. Indicate in `hal/targets.json` that your target has an entropy source.
 
 The next two sections explain how to do this.
 
-## How to implement the hardware poll
+## How to implement the TRNG api
 
-The `mbedtls_hardware_poll` function is declared in `entropy_poll.h`:
+The implementation of this interface needs to be in the HAL.
```

Comment:
What do you mean by 'in the HAL'? Do you mean it has to be in the HAL layer? Or in a specific directory?